lib/get-config.js

```diff
@@ -283,6 +286,44 @@ function processIgnores(config) {
   config.ignore = config.ignore.map((pattern) => micromatch.matcher(pattern));
 }
 
+/**
+ * we were passed a rule, add the rule being passed in, to the config.
+ * ex:
+ * rule:severity
+ * no-implicit-this:["error", { "allow": ["some-helper"] }]
+ */
+function getRuleFromString(rule) {
+  const indexOfSeparator = rule.indexOf(':') + 1;
+
+  // we have to split based on the index of the first instance of the separator because the separator could exist in the second half of the rule
+  const name = rule.substring(0, indexOfSeparator - 1);
+  const ruleConfig = rule.substring(indexOfSeparator);
+
+  // if we get passed an object that looks like no-implicit-this:["error", { "allow": ["some-helper"] }] we should split it accordingly
+  if (!_determineConfigForSeverity(ruleConfig)) {
+    const options = JSON.parse(ruleConfig);
+
+    if (Array.isArray(options)) {
+      const { severity, config: defaultConfig } = _determineConfigForSeverity(options[0]);
+      const config = options[1];
+
+      return {
+        name,
+        severity,
+        config: config || defaultConfig,
+      };
+    }
+  }
+
+  const { severity, config } = _determineConfigForSeverity(ruleConfig);
+
+  return {
+    name,
+    severity,
+    config,
+  };
```

Comment:
Thoughts on doing something like:

```suggestion
if (ruleConfig.startsWith('[')) {
  ruleConfig = JSON.parse(ruleConfig);
}

const config = determineRuleConfig(ruleConfig);

return { name, config };
```

Specifically:

* `determineRuleConfig` already handles dealing with an array or a string and returns the proper `severity`/`config` objects back. This allows us to avoid multiple calls to `_determineConfigForSeverity` (one for detecting, then another for actual processing).
* Avoid destructuring the object just to restructure it. The current destructuring/restructuring will cause errors if we ever decide to change the value returned by `determineRuleConfig` (which was intended as the single place that deals with converting userland config into our internal structure).